scripts/west_commands/runners/nrf_common.py

```diff
@@ -260,15 +273,35 @@ def program_hex(self):
             if self.hex_refers_region(xip_start, xip_end):
                 qspi_erase_opt = 'ERASE_ALL'
 
+        if self.erase_uicrs and self.family in UICR_RANGES:
+            for core, range in UICR_RANGES[self.family].items():
+                self.exec_op('erasepage', defer=True, core=core, page=range[0])
+
         # What tool commands do we need to flash this target?
         if self.family == 'NRF53_FAMILY':
             # nRF53 requires special treatment due to the extra coprocessor.
             self.program_hex_nrf53(erase_arg, qspi_erase_opt)
+        elif self.family == 'NRF54H_FAMILY':
+            self.program_hex_nrf54h()
         else:
             self.op_program(self.hex_, erase_arg, qspi_erase_opt, defer=True)
 
         self.flush(force=False)
 
+    def program_hex_nrf54h(self):
+        if self.build_conf.getboolean('CONFIG_SOC_NRF54H20_ENG0_CPUAPP'):
+            core = 'NRFDL_DEVICE_CORE_APPLICATION'
+        elif self.build_conf.getboolean('CONFIG_SOC_NRF54H20_ENG0_CPURAD'):
+            core = 'NRFDL_DEVICE_CORE_NETWORK'
+        else:
+            raise NotImplementedError("Unsupported core")
```

Comment:
I don't know how nrfjprog works, but in theory, choosing between app and radio shouldn't matter much when flashing, because neither domain has write access to MRAM by default. I think this is mediated by secure domain in either case.

I wonder if it works to flash everything via `CP_APPLICATION`. That would simplify supporting PPR.

Reply:
This is the domain to core table we have in -next as of today:

```
_DOM2COP = {
    "application": "CP_APPLICATION",
    "radiocore": "CP_NETWORK",
    "ppr": "CP_APPLICATION",
    "secure": "CP_SECURE",
    "sysctrl": "CP_APPLICATION",
}
```

So as you can see, ppr for example uses CP_APPLICATION. I'm not sure if core can be obviated or not, probably a question for nrfjprog authors?

Reply:
Based on my own testing, omitting `--coprocessor` is fine in common cases. I suggest doing that, and removing the `NotImplementedError`. For special cases, we can add this out of tree:
```cmake
board_runner_args(nrfjprog "--tool-opt=--coprocessor (...)")
```

Reply:
thanks for the investigation, it looks like nrf54h20 flashing gets easier from the initial "specs". I've simplified the patch, and checked with the latest changes, cpuapp boots fine and UICR is erased/flashed.